pkg/controller/nebulabackup/nebula_backup_manager.go

```diff
@@ -118,65 +140,346 @@ func (bm *backupManager) Sync(backup *v1alpha1.NebulaBackup) error {
+	return &corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      backup.Name,
-			Namespace: backup.Namespace,
+			Name:      nbObjectKey.Name,
+			Namespace: nbObjectKey.Namespace,
+			Labels: map[string]string{
+				"backup-name":      backup.Name,
+				"backup-namespace": backup.Namespace,
+			},
 		},
-		Spec: batchv1.JobSpec{
-			Parallelism:  pointer.Int32(1),
-			Completions:  pointer.Int32(1),
-			BackoffLimit: pointer.Int32(0),
-			Template: corev1.PodTemplateSpec{
-				Spec: corev1.PodSpec{
-					ImagePullSecrets: backup.Spec.ImagePullSecrets,
-					Containers: []corev1.Container{
-						{
-							Name:            "backup",
-							Image:           backup.Spec.ToolImage,
-							ImagePullPolicy: backup.Spec.ImagePullPolicy,
-							Env: []corev1.EnvVar{
-								{
-									Name: EnvS3AccessKeyName,
-									ValueFrom: &corev1.EnvVarSource{
-										SecretKeyRef: &corev1.SecretKeySelector{
-											LocalObjectReference: corev1.LocalObjectReference{
-												Name: backup.Spec.BR.S3.SecretName,
+		Data: map[string]string{ // TODO: remove x. x means no need to configure, but it is a required field in the program.
+			"backup-cleanup.sh": `#!/bin/bash
+ACCESS_KEY=$S3_ACCESS_KEY
+SECRET_KEY=$S3_SECRET_KEY
+
+function usage {
+    echo "usage $0 [-h] < -m meta_address, -s storage, -r s3.region, -e s3.endpoint, -b max_backups, -t max_reserved_time >"
+    echo "-h displays this message"
+    echo "-m specifies the meta server address"
+    echo "-s specifies the backup target url"
+    echo "-r specifies the s3 bucket region"
+    echo "-e specifies the s3 endpoint"
+    echo "-b specifies the maximum number of backups"
+    echo "-t spefifies the maximum reserved time for backups"
+}
+
+function convert_to_sec {
+    local to_convert="$1"
+
+    unit=${to_convert: -1}
+    num=${to_convert::-1}
+
+    case "${unit}" in
+        s)
+            echo $num
+            ;;
+        m)
+            echo "$((num * 60))"
+            ;;
+        h)
+            echo "$((num * 60 * 60))"  
+            ;;
+        d)
+            echo "$((num * 24 * 60 * 60))"
+            ;;
+        w)
+            echo "$((num * 7 * 24 * 60 * 60))"
+            ;;
+        M)
+            echo "$((num * 30 * 24 * 60 * 60))"
+            ;;
+        Y)
+            echo "$((num * 365 * 24 * 60 * 60))"
+            ;;
+        "0" | "1" | "2" | "3" | "4" | "5" | "6" | "7" | "8" | "9")
+            echo $to_convert
+            ;;
+        *)
+            echo "Invalid value $to_convert"
+            ;;
+    esac
+}
+
+
+while getopts ':hm:s:r:e:b:t:' arg; do
+  case "${arg}" in
+    h) 
+        usage
+        exit 0
+        ;;
+    m) 
+        meta_address="${OPTARG}"
+        ;;
+    s) 
+        storage="${OPTARG}" 
+        ;;
+    r) 
+        s3_region="${OPTARG}" 
+        ;;
+    e)
+        s3_endpoint="${OPTARG}"
+        ;;
+    b)
+        max_backups="${OPTARG}"
+        ;;
+    t)
+        max_reserved_time="${OPTARG}"
```

Comment:
I think inject the value through the environment is simpler

Reply:
Ok, that makes sense. Will fix this.

Reply:
Fixed